library/hls/src/main/java/com/google/android/exoplayer2/source/hls/HlsSampleStreamWrapper.java

```diff
@@ -671,7 +671,7 @@ public long getBufferedPositionUs() {
           lastMediaChunk.isLoadCompleted()
               ? lastMediaChunk
               : mediaChunks.size() > 1 ? mediaChunks.get(mediaChunks.size() - 2) : null;
-      if (lastCompletedMediaChunk != null) {
+      if (lastCompletedMediaChunk != null && lastCompletedMediaChunk.hasSamples()) {
         bufferedPositionUs = max(bufferedPositionUs, lastCompletedMediaChunk.endTimeUs);
```

Comment:
Idea is pretty basic, The sample stream wrapper should not proactively report the last loaded chunk (segment) as having buffered time if it is known to have no samples (e.g. an EXT-X-GAP tag has no media to load, so it cannot add to the buffered time).

This way, the largest queued timestamp (below on line 679) will take precedence for the buffered time.

Reply:
@christosts Why do we ever care about the `endTimeUs`, at least as long as `sampleQueuesBuilt`